### PR TITLE
[RLlib] Fix saving and loading algos and policies docs example for RL Modules

### PR DESCRIPTION
**Pull request by ArturNiederfahrenhorst**
## Why are these changes needed?

Makes it so that the docs example for saving and loading algos and policies works with RLModules.
The underlying issue is that RLModules don't support ONNX format today.

### CODE REVIEW

File: rllib/examples/documentation/saving_and_loading_algos_and_policies.py [hunk 1]
```diff
@@ -231,6 +231,7 @@ def new_policy_mapping_fn(agent_id, episode, worker, **kwargs):
 # ppo_config.framework("torch")
 
 # Create the Algorithm and train one iteration.
+# We disable the RLModules and Learner APIs here, since
```
Comment (kouroshHakha) [inline rllib/examples/documentation/saving_and_loading_algos_and_policies.py:234]: This comment is no applicable anymore I guess? the disabling happens further below.

File: rllib/examples/documentation/saving_and_loading_algos_and_policies.py [hunk 2]
```diff
@@ -295,9 +296,14 @@ def new_policy_mapping_fn(agent_id, episode, worker, **kwargs):
 
 # __export-models-3-end__
 
+# Exporting Policies in ONNX format is only supported without RLModules
+# TODO (Artur): Switch this to the RLModule API once it is supported.
+ppo_config = (
+    PPOConfig().environment("Pendulum-v1").checkpointing(export_native_model_files=True)
+).rl_module(_enable_rl_module_api=True).training(_enable_learner_api=True)
```
Comment (kouroshHakha) [inline rllib/examples/documentation/saving_and_loading_algos_and_policies.py:303]: Don't you have to construct the ppo_policy again from this new config?

File: rllib/examples/documentation/saving_and_loading_algos_and_policies.py (hunk 1 above)
Comment (ArturNiederfahrenhorst) [reply rllib/examples/documentation/saving_and_loading_algos_and_policies.py:234]: Whoops. Yeah wanted to go over these PRs before passing them into review lol. Thanks

**Comment by ArturNiederfahrenhorst** [comment]
Screenshot 2023-03-30 at 14 11 45

Test is running, but appears to be flakey so I'm making this 300s->900s.
This test is not about performance at all and we don't see regressions here so it should not matter.

**Comment by ArturNiederfahrenhorst** [comment]
After this, I'll post a screenshot of the test succeeding and will revert the rl module flags.

**Comment by ArturNiederfahrenhorst** [comment]
Screenshot 2023-03-31 at 09 38 32

**Review by kouroshHakha — APPROVED** [review]

**Comment by ArturNiederfahrenhorst** [comment]
@kouroshHakha Just to be clear, not 2.4 critical, eh? 🙂

**Review by gjoliver — APPROVED** [review]

**Pull request merged**